### PR TITLE
feat: add discrete saving functionality

### PR DESCRIPTION
**Pull request by AayushSabharwal**
## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

File: src/callbacks.jl [hunk 1]
```diff
@@ -611,6 +613,9 @@ end
         end
         integrator.u_modified = true
         callback.affect!(integrator)
+
+        SciMLBase.save_discrete_parameters_after_callback(
+            integrator.sol, parameter_values(integrator), current_time(integrator))
```
Comment (ChrisRackauckas) [inline src/callbacks.jl:618]: so it would interpolate separately?

Comment (AayushSabharwal) [reply src/callbacks.jl:618]: Interpolation is independent of this. I haven't gotten around to it yet

Comment (AayushSabharwal) [reply src/callbacks.jl:618]: Interpolation is implemented in https://github.com/SciML/SciMLBase.jl/pull/645

**Comment by AayushSabharwal** [comment]
Requires https://github.com/SciML/SciMLBase.jl/pull/645

**Comment by AayushSabharwal** [comment]
This PR is not necessary anymore

**Pull request closed without merging**